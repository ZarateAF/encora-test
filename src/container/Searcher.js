import { useState } from "react";
import { Button, Input } from "../components";

export const Searcher = ({ onClick }) => {
  const [topic, setTopic] = useState("");

  const handleClick = () => {
    if(topic !== "") onClick(topic)
  }

  return (
    <div className="search">
      <Input
        type="text"
        placeholder="Topic..."
        onChange={(e) => setTopic(e.target.value)}
        value={topic}
      />
      <Button onClick={handleClick}>Search</Button>
    </div>
  );
};
