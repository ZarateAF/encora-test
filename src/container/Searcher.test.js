import renderer from "react-test-renderer";
import { Searcher } from "./Searcher";

it("renders correctly", () => {
  const tree = renderer.create(<Searcher onClick={(a) => a} />).toJSON();
  expect(tree).toMatchSnapshot();
});
