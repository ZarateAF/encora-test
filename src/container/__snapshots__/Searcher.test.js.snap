// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search"
>
  <input
    className="sc-gsnTZi DXVWH"
    onChange={[Function]}
    placeholder="Topic..."
    type="text"
    value=""
  />
  <button
    className="sc-bczRLJ hrGClw"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
